starlette/testclient.py

```diff
@@ -430,6 +435,16 @@ def __init__(
         self.app = asgi_app
         self.base_url = base_url
 
+    @contextlib.contextmanager
+    def _portal_factory(
+        self,
+    ) -> typing.Generator[anyio.abc.BlockingPortal, None, None]:
```

Comment:
```suggestion
    ) -> typing.Iterator[anyio.abc.BlockingPortal]:
```

Reply:
mypy still passes though?

Or is https://docs.python.org/3/library/typing.html#typing.ContextManager more appropriate?

Reply:
Nevermind on typing.ContextManager, derp

Reply:
The typeshed definition for `contextmanager` takes an `Iterabe`: https://github.com/python/typeshed/blob/master/stdlib/contextlib.pyi#L37

Reply:
it's a bug: https://github.com/python/typeshed/issues/2772

Reply:
Alrighty, never-mind then